runtime/sema/check_assignment.go

```diff
@@ -370,3 +370,19 @@ func (checker *Checker) visitMemberExpressionAssignment(
 
 	return member.TypeAnnotation.Type
 }
+
+func IsValidAssignmentTargetExpression(expression ast.Expression) bool {
+	switch expression := expression.(type) {
+	case *ast.IdentifierExpression:
+		return true
+
+	case *ast.IndexExpression:
+		return IsValidAssignmentTargetExpression(expression.TargetExpression)
+
+	case *ast.MemberExpression:
+		return IsValidAssignmentTargetExpression(expression.Expression)
```

Comment:
what's the difference between `expression.TargetExpression` and `expression.Expression`?

Reply:
Not sure I understand the question. `IndexExpression.TargetExpression` is the expression that is indexed into, `MemberExpression.Expression` is the expression that is accessed

Reply:
nvm, I misunderstood the logic. Looks good to me.